enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/ParameterSettings.tsx

```diff
@@ -0,0 +1,101 @@
+import { useDebouncedCallback } from "@mantine/hooks";
+import { useCallback } from "react";
+import { P, match } from "ts-pattern";
+import { t } from "ttag";
+
+import { Stack, Text, TextInput } from "metabase/ui";
+
+import { useSdkIframeEmbedSetupContext } from "../context";
+import { useHideParameter } from "../hooks";
+
+import { ParameterVisibilityToggle } from "./ParameterVisibilityToggle";
+
+export const ParameterSettings = () => {
+  const {
+    experience,
+    settings,
+    updateSettings,
+    availableParameters,
+    isLoadingParameters,
+  } = useSdkIframeEmbedSetupContext();
+
+  const { isParameterHidden, toggleParameterVisibility } = useHideParameter();
+
+  const isQuestionOrDashboardEmbed =
+    !!settings.questionId || !!settings.dashboardId;
+
+  const updateInitialParameterValue = useDebouncedCallback(
+    useCallback(
+      (paramId: string, value: string) => {
+        if (settings.dashboardId) {
```

Comment:
For type narrowing. We cannot use `experience === "dashboard"` here as it does not narrow the type.